pkg/services/searchV2/extract/dashboard.go

```diff
@@ -271,6 +272,29 @@ func ReadDashboard(stream io.Reader, lookup DatasourceLookup) (*DashboardInfo, e
 	return dash, iter.Error
 }
 
+func filterOutSpecialDatasources(dash *DashboardInfo) {
+	for i, panel := range dash.Panels {
+
+		var dsRefs []DataSourceRef
+
+		// partition into actual datasource references and variables
+		for _, ds := range panel.Datasource {
+			switch ds.UID {
+			case "-- Mixed --":
+				// The actual datasources used as targets will remain
+				continue
+			case "-- Dashboard --":
```

Comment:
I think we should keep this one (just like "-- Grafana --"), but the permissions model is obviously complicated!

This is an important enough distinction that we actually will want to search (and report) on where folks are using shared dashboard queries

Reply:
@ryantxu  The reason I filtered it out is that `-- Dashboard --` is just not a reference to a backend entity called `Datasource`, ie. it's not `DatasourceRef`... even if I squint hard enough like I did with `-- Grafana --` :D 

I find exceptions such as this cause a lot of issues/mental overhead downstream, RBAC/permissions being just one of (and first) them.

This would be simpler to handle if the model looked more like the following, where panel's/datasource's data can be populated either via a "ds query" or some other means:

```ts
type panel = {
    ...,
    data: {
        "my-data-ref-1": {
            type: "datasource_query",
            ...
        },
        "my-data-ref-2": {
            type: "dashboard_panel",
            panelId: 5,
        },
    }

}
```

I understand that, in general, we want to retain this data - wdyt about either
1. populating the datasource array of panels with `-- Dashboard --` ds with datasources of the referenced panel rather than keeping it empty
2. adding an extra field called `referenced_panel` (or similar naming) that would contain the referenced `panelId`

or both?